engine/src/main/java/org/pentaho/di/trans/ael/websocket/DaemonMessagesClientEndpoint.java

```diff
@@ -0,0 +1,212 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * *****************************************************************************
+ */
+package org.pentaho.di.trans.ael.websocket;
+
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.variables.Variables;
+import org.pentaho.di.engine.api.remote.ExecutionRequest;
+import org.pentaho.di.engine.api.remote.Message;
+import org.pentaho.di.engine.api.remote.MessageDecoder;
+import org.pentaho.di.engine.api.remote.MessageEncoder;
+import org.pentaho.di.engine.api.remote.StopMessage;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.trans.ael.websocket.exception.MessageEventFireEventException;
+
+import javax.websocket.ClientEndpointConfig;
+import javax.websocket.ContainerProvider;
+import javax.websocket.CloseReason;
+import javax.websocket.Endpoint;
+import javax.websocket.EndpointConfig;
+import javax.websocket.MessageHandler;
+import javax.websocket.Session;
+import javax.websocket.WebSocketContainer;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Created by fcamara on 8/17/17.
+ */
+
+public class DaemonMessagesClientEndpoint extends Endpoint {
+  /**
+   * The package name, used for internationalization of messages.
+   */
+  private static Class<?> PKG = DaemonMessagesClientEndpoint.class; // for i18n purposes
+  //i18n messages
+  private static final String EXCEPTION_SESSION_CLOSED = "DaemonMessagesClientEndpoint.Exception.SessionIsClosed";
+
+  private static final String KETTLE_AEL_PDI_DAEMON_PRINCIPAL = "KETTLE_AEL_PDI_DAEMON_PRINCIPAL";
+  private static final String KETTLE_AEL_PDI_DAEMON_KEYTAB = "KETTLE_AEL_PDI_DAEMON_KEYTAB";
+  private static final String KETTLE_AEL_PDI_DAEMON_CONTEXT_REUSE = "KETTLE_AEL_PDI_DAEMON_CONTEXT_REUSE";
+  private static final String Y_LWC = "y";
+  private static final int MAX_TXT_MSG_BUF_SIZE = 500000;
+  private static final int MAX_BIN_MSG_BUF_SIZE = 500000;
+
+  private static final String URL_TEMPLATE = "%s://%s:%s/execution";
+  private static final String PRFX_WS = "ws";
+  private static final String PRFX_WS_SSL = "wss";
+  private final MessageEventService messageEventService;
+  private Session userSession = null;
+  private String principal = null;
+  private String keytab = null;
+  private boolean reuseSparkContext = false;
+  //only one stop message
+  private AtomicBoolean alReadySendedStopMessage = new AtomicBoolean( false );
+
+  public DaemonMessagesClientEndpoint( String host, int port, boolean ssl,
+                                       MessageEventService messageEventService ) throws KettleException {
+    try {
+      setAuthProperties();
+
+      String url = String.format( URL_TEMPLATE, ( ssl ? PRFX_WS_SSL : PRFX_WS ), host, port );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: port (declared at line 80). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 85. Consider formatting this data with &#39;%d&#39; instead.